src/groupchat.c

```diff
@@ -301,6 +352,65 @@ static void groupchat_onGroupTitleChange(ToxWindow *self, Tox *m, uint32_t group
     write_to_log(tmp_event, nick, ctx->log, true);
 }
 
+/* Puts `(NameListEntry *)`s in `entries` for each matched peer, up to a maximum
+ * of `maxpeers`.
+ * Maches each peer whose name or pubkey begins with `prefix`.
+ * If `prefix` is exactly the pubkey of a peer, matches only that peer.
+ * return number of entries placed in `entries`.
+ */
+uint32_t get_name_list_entries_by_prefix(uint32_t groupnumber, const char *prefix, NameListEntry **entries,
+        uint32_t maxpeers)
+{
+    GroupChat *chat = &groupchats[groupnumber];
+
+    if (!chat->active) {
+        return 0;
+    }
+
+    const size_t len = strlen(prefix);
+
+    if (len == 2 * TOX_PUBLIC_KEY_SIZE) {
+        for (uint32_t i = 0; i < chat->num_peers; ++i) {
+            if (strcasecmp(prefix, chat->name_list[i].pubkey_str) == 0) {
+                entries[0] = &chat->name_list[i];
+                return 1;
+            }
+        }
+    }
+
+    uint32_t n = 0;
+
+    for (uint32_t i = 0; i < chat->num_peers; ++i) {
+        if (strncmp(prefix, chat->name_list[i].name, len) == 0
```

Comment:
I'd give a name to `&chat->name_list[i]` here (maybe also above), but it's borderline. What do you think?